### PR TITLE
GH-38541: [MATLAB] Add remaining tests for arrow tabular object display

### LINKED ISSUE #38541: [MATLAB] Add remaining tests for arrow tabular object display
### Describe the enhancement requested

The current arrow tabular object display tests only cover basic test cases. Please add more test cases to ensure its quality.

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by leihou6116**
### Rationale for this change

The current tests for arrow tabular display only covers basic use cases. To guarantee its quality, we need to add tests to cover more use cases.

### Added Test Cases
- display of 0-by-N tabular
- display of a very wide tabular to verify that the display is not truncated.
- display of hyperlink contained in the tabular data
- display class name when the datatype of a tabular variable is not array.
- display of null
- display of Unicode.

### Are these changes tested?

Yes.

See "Added test Cases" above.

### Are there any user-facing changes?

No
* Closes: #38541

### CODE REVIEW

**Comment by kevingurney** [comment]
@leihou6116 - could you please update the description of this pull request to describe what tests are being added?

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 1]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
 
             testCase.verifyEqual(actualDisplay, expectedDisplay);
         end
+
+        function veryWideTabular(testCase, TabularType)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:157]: Could you please use `UpperCamelCase` for the test names to follow the convention used in this file?

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 2]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            t = array2table([1:100;101:200],VariableNames="x"+(1:100));
+            arrowTabularObj = TabularType.FromTableFcn(t);
+            actualDisplay = evalc('disp(arrowTabularObj)');
+
+            classNameString = makeLinkString(FullClassName=TabularType.FullClassName, ...
+                ClassName=TabularType.ClassName, BoldFont=true);
+            header = compose("  Arrow %s with %s rows and %s columns:", classNameString, getNumString(2), getNumString(100));
+            schemaDisplay = ['    Schema:', newline, newline, '        '];
+            dataDisplay   = ['    First Row:', newline, newline, '        '];
+            for i = 1 : width(t)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:171]:
```suggestion
            for i = 1:width(t)
```

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 3]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            expectedDisplay = [char(header), newline, ...
+                               newline, ...
+                               schemaDisplay, newline, ...
+                               newline ...
+                               dataDisplay, newline, ...
+                               newline];
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+
+        function dataContainHyperlink(testCase, TabularType)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:189]:
```suggestion
        function DataContainsHyperlink(testCase, TabularType)
```

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 4]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            % No matter whether hotlinks is turned on or off, the data text
+            % containing hyperlink is not impacted and is displayed well.
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:191]:
```suggestion
            % containing hyperlink is not impacted and is displayed as expected.
```

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 5]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            import arrow.internal.test.display.makeLinkString
+
+            hLink1 = string('<a href="http://www.foo.com">foo</a>');
+            hLink2 = string('<a href="http://www.foo.com">another foo</a>');
+            t = table(["a";"bcd";missing;""],[hLink1;hLink2;hLink1;hLink2],[NaN;1;2;3],'VariableNames',["Description", "Link_to_doc", "Result"]);
+            arrowTabularObj = TabularType.FromTableFcn(t);
+
+            % The display of schema and actual tabular data always contain
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:199]:
```suggestion
            % The display of schema and actual tabular data always contains 
```

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 6]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            actualDisplay = evalc('feature(''hotlinks'',''off'');disp(arrowTabularObj)');
+            testCase.verifySubstring(actualDisplay, expectedDisplayOfData);
+
+            % hotlinks is turned on
+            actualDisplay = evalc('feature(''hotlinks'',''on'');disp(arrowTabularObj)');
+            testCase.verifySubstring(actualDisplay, expectedDisplayOfData);
+        end
+
+        function displayClassNameWhenDataIsNotArray(testCase, TabularType)
+            % When the datatype of a column is not an homogeneous array
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:228]:
```suggestion
            % When the datatype of a column is not a homogeneous array
```

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 7]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+                               '    First Row:', newline, ...
+                               newline, ...
+                               '        ', '2023-01-01 00:00:00.000000 | <Struct> | 00:00:01.000000', newline,...
+                               newline];
+
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+
+        function displayInvalidData(testCase, TabularType)
+            % Display "null" for invalid value
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:258]: For consistency with the other test cases, could you start the test case comments with something like "Verify that..."

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 8]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            import arrow.internal.test.display.makeLinkString
+
+            t = table(seconds([NaN;1]), string([missing;"a"]), string(["";"b"]),  [NaN;1], datetime(2023,1,[NaN;2]),...
+                      VariableNames=["durationVar", "stringVar1", "stringVar2", "doubleVar", "datetimeVar"]);
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:262]: Indentation is off here.

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 9]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+                               newline, ...
+                               '    First Row:', newline, ...
+                               newline, ...
+                               '        ', 'null | null | "" | null | null', newline,...
+                               newline];
+
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+
+        function unicodeDisplayWell(testCase, TabularType)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:288]:
```suggestion
        function Unicode(testCase, TabularType)
```

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 10]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+            % The current display doesn't align anything. So it should work
+            % well with displaying unicode.
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:290]:
```suggestion
            % well with displaying Unicode characters.
```

**Review by kevingurney — CHANGES_REQUESTED** [review]
Thanks for adding these tests! I just had a few minor comments.

File: matlab/test/arrow/tabular/tTabularDisplay.m (hunk 1 above)
Comment (leihou6116) [reply matlab/test/arrow/tabular/tTabularDisplay.m:157]: Good catch. Thank you. I'll update the test method name to follow the convention.

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 11]
```diff
@@ -76,6 +76,34 @@ function ZeroRowsOneColumn(testCase, TabularType)
             testCase.verifyEqual(actualDisplay, expectedDisplay);
         end
 
+        function zeroRowsMultipleColumns(testCase, TabularType)
+            % Test the display of 0xN arrow tabular object.
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tTabularDisplay.m:80]: Missing an "a" between "of" and "0xN" -> "of a 0xN"

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 12]
```diff
@@ -124,6 +153,167 @@ function ManyRowsAndColumns(testCase, TabularType)
...
+                               '    First Row:', newline, ...
+                               newline, ...
+                               '        ', 'null | null | "" | null | null', newline,...
+                               newline];
+
+            testCase.verifyEqual(actualDisplay, expectedDisplay);
+        end
+
+        function unicodeDisplayWell(testCase, TabularType)
+            % The current display doesn't align anything. So it should work
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tTabularDisplay.m:289]: This is a bit confusing. I would say the current display does not attempt to align multiple rows vertically. Therefore, it should work well displaying Unicode characters.

File: matlab/test/arrow/tabular/tTabularDisplay.m [hunk 13]
```diff
@@ -224,10 +227,10 @@ function dataContainHyperlink(testCase, TabularType)
...
         end
 
-        function displayClassNameWhenDataIsNotArray(testCase, TabularType)
-            % When the datatype of a column is not an homogeneous array
-            % type, the class name instead of the actual data will be
-            % displayed.
+        function DisplayClassNameWhenDataIsNotArray(testCase, TabularType)
+            % Verify that the class name instead of the actual data will be
+            % displayed when the datatype of a tabular variable is not a 
+            % homogeneous array type.
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tTabularDisplay.m:233]:
```suggestion
            % nested array type (e.g. StructArray or ListArray)
```

**Review by kevingurney — APPROVED** [review]

**Pull request merged**